docker/daemon.go

```diff
@@ -79,14 +91,37 @@ func migrateKey() (err error) {
 	return nil
 }
 
-func mainDaemon() {
-	if utils.ExperimentalBuild() {
-		logrus.Warn("Running experimental build")
+// DaemonCli represents the daemon CLI.
+type DaemonCli struct {
+	*daemon.Config
+	registryOptions *registry.Options
+}
+
+// CmdDaemon is the daemon command, called the raw arguments after `docker daemon`.
+func (cli *DaemonCli) CmdDaemon(args ...string) error {
+	if *flDaemon {
+		// allow `docker -D -d` and `docker -d -D`
+		logrus.Warn("please use 'docker daemon' instead.")
+	} else if !(commonFlags.FlagSet).IsEmpty() {
+		// deny `docker -D daemon`
+		logrus.Fatal("invalid global flags")
```

Comment:
Ignoring whether --log-level should be global or not, the error message isn't helpful because it doesn't tell the user what flag was bad.  If I had listed several of them I wouldn't know which one was wrong - especially since --log-level (per the help) is legal.  Can't we show the erroneous flag in the msg?